File: lib/java/linux/src/main/java/alphaTab/alphaSkia/AlphaSkiaLinux.java
package alphaTab.alphaSkia;

/**
 * The alphaSkia initializer for Linux.
 */
public class AlphaSkiaLinux extends AlphaSkiaJre {
    /**
     * The alphaSkia initializer for Linux.
     */
    public static final AlphaSkiaLinux INSTANCE = new AlphaSkiaLinux();

    private AlphaSkiaLinux() {
    }

    @Override
    protected String[] getJavaResources() {
        return new String[] {
            "/native/linux-" + getCurrentArchitecture() + "/libalphaskiajni.so"
        };
    }
}

File: lib/kotlin/linux/src/main/java/alphaTab/alphaSkia/AlphaSkiaLinux.kt
package alphaTab.alphaSkia

/**
 * The alphaSkia initializer for Linux.
 */
class AlphaSkiaLinux private constructor() : AlphaSkiaJre() {
    override fun getJavaResources(): Array<String> {
        return arrayOf(
            "/native/linux-" + getCurrentArchitecture() + "/libalphaskiajni.so"
        )
    }

    companion object {
        /**
         * The alphaSkia initializer for Linux.
         */
        @JvmField
        val INSTANCE = AlphaSkiaLinux()
    }
}

